e2eTest/src/main/java/com/google/android/ground/e2etest/SurveyRunnerTest.kt

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.e2etest
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Direction
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.Until
+import com.google.android.ground.R
+import com.google.android.ground.e2etest.TestConfig.GROUND_PACKAGE
+import com.google.android.ground.e2etest.TestConfig.LONG_TIMEOUT
+import com.google.android.ground.e2etest.TestConfig.SHORT_TIMEOUT
+import com.google.android.ground.e2etest.TestConfig.TEST_SURVEY_IDENTIFIER
+import com.google.android.ground.e2etest.TestConfig.TEST_SURVEY_TASKS_ADHOC
+import com.google.android.ground.model.task.Task
+import junit.framework.TestCase.fail
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class SurveyRunnerTest : AutomatorRunner {
+
+  override lateinit var device: UiDevice
+
+  private val selectSurveyTitle = stringResource(R.string.select_survey_title)
+  private val collectData = stringResource(R.string.collect_data)
+  private val newDataCollectionSite = stringResource(R.string.new_site)
+  private val close = stringResource(R.string.close)
+  private val dropPin = stringResource(R.string.drop_pin)
+  private val addPoint = stringResource(R.string.add_point)
+  private val complete = stringResource(R.string.complete_polygon)
+  private val capture = stringResource(R.string.capture)
+  private val camera = stringResource(R.string.camera)
+  private val ok = stringResource(R.string.ok)
+  private val next = stringResource(R.string.next)
+  private val done = stringResource(R.string.done)
+  private val save = stringResource(R.string.save)
+
+  @Test
+  fun run() {
+    launchGround()
+    signIn()
+    selectTestSurvey()
+    zoomIntoLocation()
+    startAdHocLoiTask()
+    fillOutTaskData(isAdHoc = true, TEST_SURVEY_TASKS_ADHOC)
+    clickSubmissionConfirmationDone()
+    startPredefinedLoiTask()
+    fillOutTaskData(isAdHoc = false, TEST_SURVEY_TASKS_ADHOC.drop(1))
+    clickSubmissionConfirmationDone()
+  }
+
+  private fun launchGround() {
```

Comment:
I wonder if we can store these UI actions in their own files so they can be reused and to keep the test free from boiler plate?

Reply:
I considered making an `AutomatorHelper` class, but it was annoying to hop between files since there was only one test. We should consider refactoring more functionality into the `AutomatorRunner` interface to share functionality between future classes, but for now, it's nicer to have everything in one file in top to bottom order of more to less abstract, e.g.:

```
    launchGround()
    signIn()
    selectTestSurvey()
    zoomIntoLocation()
    startAdHocLoiTask()
    fillOutTaskData(isAdHoc = true, TEST_SURVEY_TASKS_ADHOC)
    clickSubmissionConfirmationDone()
    startPredefinedLoiTask()
    fillOutTaskData(isAdHoc = false, TEST_SURVEY_TASKS_ADHOC.drop(1))
    clickSubmissionConfirmationDone()
```